### PR TITLE
Make the release action upload to real PyPI

### PR DESCRIPTION
**Pull request by DMRobertson**
Test was successful: see https://github.com/matrix-org/synapse-s3-storage-provider/runs/4312656424?check_suite_focus=true and https://test.pypi.org/project/synapse-s3-storage-provider/1.1.2/

### CODE REVIEW

File: setup.py
Comment (babolivier) [inline setup.py:3]: Shouldn't we follow semantic versioning like we do in other projects? In which case the version should be `1.1.0`.

Comment (babolivier) [reply setup.py:3]: Unless you're just testing stuff and I should just wait till it's up for review (github doesn't say a PR is a draft when sending a notification so I missed that status :cry: )

Comment (DMRobertson) [reply setup.py:3]: 1.1 already exists; it was chosen by the previous maintainer. I don't know if PyPI, pip etc consider these equivalent.

This change is intended to revert to the status quo after my testing on test.pypi.org. I'll can leave a comment encouraging a bump to 1.1.1 for the next version?

Comment (babolivier) [reply setup.py:3]: Makes sense, and a comment sgtm

**Comment by DMRobertson** [comment]
@squahtx, I'd be interested in your thoughts here in the light of #65!

File: .github/workflows/ci.yml
Comment (babolivier) [inline .github/workflows/ci.yml:37]:
```suggestion
    name: "Build and check Python package"
```

Not sure "Python package" is the right terminology here, but we should make it explicit we're not talking about eg Debian packaging.

File: README.md
Comment (babolivier) [inline README.md:87]:
```suggestion
3. When published, a [GitHub action](https://github.com/matrix-org/synapse-s3-storage-provider/actions) will build the package and upload to [PyPI](https://pypi.org/project/synapse-s3-storage-provider/).
```

I think it'd be more useful to tell the maintainer where to look for said action, but ymmv

File: README.md
Comment (babolivier) [inline README.md:89]: Not sure if we should write stuff about matrix.org ops in readmes, but maybe that's just me being too paranoid.

**Review by babolivier — APPROVED** [review]
lgtm otherwise

File: README.md
Comment (squahtx) [reply README.md:89]: Agree on this, this doesn't seem like the right place for details about the matrix.org deployment

**Review by squahtx — APPROVED** [review]

File: README.md
Comment (DMRobertson) [reply README.md:89]: Thanks, will revert this bit.

File: README.md
Comment (DMRobertson) [reply README.md:87]: Argh, this was meant to be the workflow itself, not a specific run of it. Thanks.

**Pull request merged**
